packages/api/elastic_migrate.py

```diff
@@ -0,0 +1,199 @@
+#!/usr/bin/python
+import os
+import json
+import psycopg2
+from psycopg2.extras import RealDictCursor
+from elasticsearch import Elasticsearch, helpers
+
+PG_HOST = os.getenv('PG_HOST', 'localhost')
+PG_PORT = os.getenv('PG_PORT', 5432)
+PG_USER = os.getenv('PG_USER', 'app_user')
+PG_PASSWORD = os.getenv('PG_PASSWORD', 'app_pass')
+PG_DB = os.getenv('PG_DB', 'omnivore')
+ES_URL = os.getenv('ES_URL', 'http://localhost:9200')
+ES_USERNAME = os.getenv('ES_USERNAME', 'elastic')
+ES_PASSWORD = os.getenv('ES_PASSWORD', 'password')
+DATA_FILE = os.getenv('DATA_FILE', 'data.json')
+BULK_SIZE = os.getenv('BULK_SIZE', 100)
+UPDATE_TIME = os.getenv('UPDATE_TIME', '2019-01-01 00:00:00')
+INDEX_SETTINGS = os.getenv('INDEX_SETTINGS', 'index_settings.json')
+
+DATETIME_FORMAT = 'YYYY-MM-DD"T"HH24:MI:SS.MS"Z"'
+QUERY = f'''
+   SELECT
+     p.id,
+     title,
+     description,
+     to_char(l.created_at, '{DATETIME_FORMAT}') as "createdAt",
+     to_char(l.updated_at, '{DATETIME_FORMAT}') as "updatedAt",
+     url,
+     hash,
+     original_html as "originalHtml",
+     content,
+     author,
+     image,
+     to_char(published_at, '{DATETIME_FORMAT}') as "publishedAt",
+     upload_file_id as "uploadFileId",
+     page_type as "pageType",
+     user_id as "userId",
+     to_char(shared_at, '{DATETIME_FORMAT}') as "sharedAt",
+     article_reading_progress as "readingProgressPercent",
+     article_reading_progress_anchor_index as "readingProgressAnchorIndex",
+     to_char(saved_at, '{DATETIME_FORMAT}') as "savedAt",
+     slug,
+     to_char(archived_at, '{DATETIME_FORMAT}') as "archivedAt"
+   FROM omnivore.pages p
+   LEFT JOIN omnivore.links l ON p.id = l.article_id
+   WHERE p.updated_at > '{UPDATE_TIME}'
+'''
+
+UPDATE_ARTICLE_SAVING_REQUEST_SQL = f'''
+UPDATE omnivore.article_saving_request
+    SET elastic_page_id = article_id
+    WHERE elastic_page_id is NULL
+        AND article_id is NOT NULL
+        AND updated_at > '{UPDATE_TIME}';
+'''
+
+UPDATE_HIGHLIGHT_SQL = f'''
```

Comment:
Do we need to do this update on other tables I see these ones in the migration (but then see the columns are removed later): 

```
ALTER TABLE omnivore.article_saving_request ADD COLUMN elastic_page_id varchar(36);
ALTER TABLE omnivore.reaction ADD COLUMN elastic_page_id varchar(36);
ALTER TABLE omnivore.highlight ADD COLUMN elastic_page_id varchar(36);
ALTER TABLE omnivore.reminders ADD COLUMN elastic_page_id varchar(36);
ALTER TABLE omnivore.abuse_report ADD COLUMN elastic_page_id varchar(36);
ALTER TABLE omnivore.content_display_report ADD COLUMN elastic_page_id varchar(36);
ALTER TABLE omnivore.link_share_info ADD COLUMN elastic_page_id varchar(36);
```

Reply:
We probably do not need to do because these tables are not being used for now @jacksonh